rust/Cargo.lock

```diff
@@ -38,59 +392,1418 @@ dependencies = [
 ]
 
 [[package]]
-name = "inc"
-version = "0.1.0"
+name = "digest"
+version = "0.9.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "d3dd60d1080a57a05ab032377049e0591415d2b31afd7028356dbf3cc6dcb066"
 dependencies = [
- "cxx",
+ "generic-array",
 ]
 
 [[package]]
-name = "link-cplusplus"
-version = "1.0.6"
+name = "directories-next"
+version = "2.0.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "339ee130d97a610ea5a5872d2bbb130fdf68884ff09d3028b81bec8a1ac23bbc"
+dependencies = [
+ "cfg-if",
+ "dirs-sys-next",
+]
+
+[[package]]
+name = "dirs"
+version = "4.0.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "ca3aa72a6f96ea37bbc5aa912f6788242832f75369bdfdadcb0e38423f100059"
+dependencies = [
+ "dirs-sys",
+]
+
+[[package]]
+name = "dirs-sys"
+version = "0.3.7"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "1b1d1d91c932ef41c0f2663aa8b0ca0342d444d842c06914aa0a7e352d0bada6"
+dependencies = [
+ "libc",
+ "redox_users",
+ "winapi",
+]
+
+[[package]]
+name = "dirs-sys-next"
+version = "0.1.2"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "4ebda144c4fe02d1f7ea1a7d9641b6fc6b580adcfa024ae48797ecdeb6825b4d"
+dependencies = [
+ "libc",
+ "redox_users",
+ "winapi",
+]
+
+[[package]]
+name = "either"
+version = "1.8.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "90e5c1c8368803113bf0c9584fc495a58b86dc8a29edbf8fe877d21d9507e797"
+
+[[package]]
+name = "env_logger"
+version = "0.9.0"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "0b2cf0344971ee6c64c31be0d530793fba457d322dfec2810c453d0ef228f9c3"
+dependencies = [
+ "atty",
+ "humantime",
+ "log",
+ "regex",
+ "termcolor",
+]
+
+[[package]]
+name = "errno"
+version = "0.2.8"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "f639046355ee4f37944e44f60642c6f3a7efa3cf6b78c78a0d989a8ce6c396a1"
+dependencies = [
+ "errno-dragonfly",
+ "libc",
+ "winapi",
+]
```

Comment:
Does rust have the equivalent of ccache that can store cached build results outside the build directory? So that `rm -rf build; ninja` will be fast?

I'm thinking of our builders which start from a clean directory each time.

Reply:
Mozilla's sccache works with rust: https://github.com/mozilla/sccache . Disclaimer: I have never used it personally so I can't tell how well it works in practice.

Reply:
Ok. Let's start using it when we see rust compilation times having an impact.

Reply:
The `incremental` option is supposed to help with rebuild times, but it does not help with the `rm -rf build; ninja` case and I only enabled it for the `dev` mode, as it might make the final binary less optimized.
I might just try this `sccache` because in the case of `rm -rf build; ninja` the rust compilation is very impactful (although this could have been expected, as rust is the only thing that is being rebuilt then)

Reply:
I tried [sccache](https://github.com/mozilla/sccache) and it helps a lot with the `rm -rf build; ninja` case. I don't expect we want to add it as one of our dependencies, but explain how to use it in some docs.  I'll need to research it a bit more so maybe we can do this as a follow-up

Reply:
We install ccache into the dbuild environment so we may as well install sscscscscache. Certainly it could be in a separate pull request.

Reply:
I see it even supports C++, so if it's better in some way, we can replace ccache.